core/src/main/scala/org/apache/spark/util/DependencyUtils.scala

```diff
@@ -150,36 +150,36 @@ private[spark] object DependencyUtils extends Logging {
     resolveMavenDependencies(
       transitive,
       exclusionList,
-      authority,
+      Some(authority),
       ivyProperties.repositories,
       ivyProperties.ivyRepoPath,
-      Option(ivyProperties.ivySettingsPath)
+      ivyProperties.ivySettingsPath
     )
   }
 
   def resolveMavenDependencies(
       packagesTransitive: Boolean,
-      packagesExclusions: String,
-      packages: String,
-      repositories: String,
-      ivyRepoPath: String,
+      packagesExclusions: Option[String],
+      packages: Option[String],
+      repositories: Option[String],
+      ivyRepoPath: Option[String],
       ivySettingsPath: Option[String]): Seq[String] = {
     val exclusions: Seq[String] =
-      if (!StringUtils.isBlank(packagesExclusions)) {
-        packagesExclusions.split(",")
+      if (packagesExclusions.nonEmpty) {
+        packagesExclusions.map(_.split(",")).get
       } else {
         Nil
       }
```

Comment:
Better to use idiomatic Option processing, either a `match` or `packagesExclusions.map(_.split(",")).getOrElse(Nil)`

Reply:
> Better to use idiomatic Option processing, either a `match` or `packagesExclusions.map(_.split(",")).getOrElse(Nil)`

DOne